### DISCUSSION TITLE
I wrote a Galaxian clone

### DISCUSSION THREAD

**Discussion by vruumllc** [picocomputer/community#66] [Q&A]
![star_swarms](https://github.com/picocomputer/community/assets/145504135/fb206419-e335-4af2-b636-589c69d4c2af)

https://github.com/vruumllc/RP6502-Star-Swarms

Here's a first draft of my new game, Star Swarms, which is a clone of the old game Galaxians.
It still needs work -- aliens are too dumb, for instance -- but I wanted to get it out there before I disappear for 2 weeks on vacation.
It serves as a great example of what you can do with asset management, tiles, and affine sprites on the RP6502. Enjoy!

Keys:
--------
Left, Right arrows -- move ship
Up, Down, Space -- fire missile
P -- Pause
R -- Restart
ESC -- Quit

**Comment by WojciechGw** [comment]
any ideas ? hardware failure ?
![1702587410832](https://github.com/picocomputer/community/assets/7420761/045a7c7b-e5c3-4153-be46-a513b5974467)

**Reply by rumbledethumps** [reply]
Use "set vga 0" for this game.

**Reply by WojciechGw** [reply]
thank You :) better but still with issues ...
![1702588393502](https://github.com/picocomputer/community/assets/7420761/2a4ce7c4-6084-435f-af1e-a600d7b2903f)

**Reply by vruumllc** [reply]
Odd, it works fine with my LCD on both "set vga 0" (640x480) and set vga 1 (640x480 and 1280x720).

**Reply by rumbledethumps** [reply]
Maybe you're running a debug build of VGA. This half-blue is what happens when authors push things to the edge of performance.

**Reply by rumbledethumps** [reply]
It fails like this on "set vga 2" for me too. But 0 seems fine. (edit changed to 2)

**Reply by vruumllc** [reply]
Interesting. I'll give that a try here and see what happens.
I am so impressed with the speed of the game! Way better than the antique DOS Atarisoft version I have.

**Reply by WojciechGw** [reply]
I went back to version 0.4 of rp6502_vga.uf2 from self-compiled and now it is ok
![1702588869937](https://github.com/picocomputer/community/assets/7420761/80246366-4c8f-4800-9843-6137ac166880)

**Reply by vruumllc** [reply]
Hmm, set vga 2 doesn't work on my LCD -- it is 640x480 max.

**Reply by WojciechGw** [reply]
effect with 0.4 firmware and "set vga 2"
![1702589139131](https://github.com/picocomputer/community/assets/7420761/3949807f-06f0-46cb-80d3-e5f3a75494bb)

**Reply by vruumllc** [reply]
Yes, I guess I have crammed as much as I can in the vsync loop. 8^)
I have seem this problem if I add a printf() for debugging in certain routines.

**Reply by WojciechGw** [reply]
anyway - awesome job :) thank You
have a good time on vacation :)

**Reply by vruumllc** [reply]
Thanks!

**Reply by netzerohero** [reply]
Nice job, so cool!  (Thanks for your contributions! - I've been reading your graphics c-code / library, for understanding.  Then distilling to the most-minimal C in an attempt to study the assembly-code to put a single-dot on the screen via BASIC, LOL.)

**Reply by vruumllc** [reply]
Great! That’s what I would do too. Good luck

**Comment by n9jcv** [comment]
WOW @vruumllc 
I dowoaded and ran this.  It must have been a huge effort.  Thank You for sharing.  I hope at some point to be able to use the source for my education about this.

**Reply by vruumllc** [reply]
You are so kind. Thanks!

**Comment by michaelsarr** [comment]
very cool, gracias

**Comment by vruumllc** [comment]
I finally added Star Swarms to [my GitHub repositories](https://github.com/vruumllc), and have updated the first message in this thread accordingly.
